### ISSUE TITLE
Best way to load multiple machine learning models to be served by FastAPI

### ISSUE THREAD

**Issue by ahmedbesbes** [tiangolo/fastapi#3418]
Hello everyone, 

Amazing package: I think everyone pretty much agrees on this!

Here's my problem: I want to build a NER API that'd use many Spacy models (of different sizes, languages)

I was wondering what was the best practice of loading these models?

What I've done so far is loading them in a dictionary at the startup of the app. But, for some reason, I feel this is not the most efficient way: if you start FastAPI with N workers, the models will (presumably) be loaded N times which may eat up all your RAM. 

Is there a better solution?

Thanks!

**Comment by frankie567** [comment]
Indeed, by default, Gunicorn will not preload anything before spawning workers and each one will load the model into memory. 

What you can try is the [`--preload` option of Gunicorn](https://docs.gunicorn.org/en/stable/settings.html#preload-app), which will load all the application code before forking the workers. Hence, your model will be loaded only once. Test this carefully though, because weird things could happen if workers are producing side-effects in the now shared memory space.

**Comment by raphaelauv** [comment]
@ahmedbesbes could you close your issue , thanks
